### ISSUE TITLE
failing to load initial state, but actually is loading initial state, but then causing other issues

### ISSUE THREAD

**Issue by faceyspacey** [michaelcontento/redux-storage#130]
For whatever reasons, the `catch` statement of the promise returned by `load` is being called, but the previous state is in fact loading. That's not the real issue that's causing me problems though--Redux's connect `mapStateToProps` functions aren't being called again when the state changes. I've pintpointed this as a Redux Storage issue because when I remove the `load` call of Redux Storage, everything works. 

So my thinking is that it's "failing" not so much in that the loading of previous state fails, but failing in some other regards with how it interfaces with the rest of Redux. What are the common reasons the promise's `catch` would be triggered?

**Comment by ferologics** [comment]
same issue here, it started randomly, i didn't touch the configure store or anything, just added more props  and logic that shouldn't relate to loading the previous state

**Comment by ferologics** [comment]
This is the line that is fucking the loading of the previous state up

`{(lastMessage.body.length > 30) ? lastMessage.body.substring(0, 30) + '...' : lastMessage.body}`

here is the project [link](https://github.com/project-xband/android/) if you want more  context

**Comment by ferologics** [comment]
it appears to be the whole [ChatCell](https://github.com/project-xband/android/blob/master/src/components/ChatCell.js) component, not sure what I'm doing wrong there.

**Comment by ferologics** [comment]
I believe it has to do with something being undefined in the store maybe

`{(!lastMessage.body === "undefined" && lastMessage.body.length > 30) ? lastMessage.body.substring(0, 30) + '...' : lastMessage.body}`

I've changed the breaking line to this and now it loads the state easily

**Comment by faceyspacey** [comment]
I found the problem also to be unrelated to redux storage. Basically I found that if some other task was intensive, redux storage wouldn't load. For me, it was `immediatelyResetRouteStack` on ex-navigator before the state was ready. No exceptions or errors were thrown, but it still broke redux storage.

In short, the solution is to defer rendering until it's loaded, which is what you should be doing anyway. If you're properly using redux storage, you have the entire last state of the app before the app even opens--you should be using it. Anything rendered without it besides initial loading screens would be a waste to load because you're likely to re-render it all with the updated store you get once redux storage loads it.

In conclusion, give redux storage clean slate to do its work loading the previous state, i.e. don't render much.

**Comment by michaelcontento** [comment]
👍  for @faceyspacey's conclusion.

I'll go ahead and close this issue for now. If there is a real bug in redux-storage or a engine, please open a new issue with some details how to reproduce the flaw.

**Comment by faceyspacey** [comment]
@michaelcontento i think it needs more error messaging because a lot of things can cause previous state to fail to be load, but it doesn't tell you why. Can we throw some more detailed exceptions throughout the code, especially on initial load?
